pallets/connectors/src/message.rs

```diff
@@ -0,0 +1,276 @@
+use cfg_primitives::Moment;
+use sp_std::{vec, vec::Vec};
+
+use crate::*;
+
+#[derive(Decode, Clone, PartialEq, TypeInfo)]
+#[cfg_attr(feature = "std", derive(Debug))]
+pub enum Message<Domain, PoolId, TrancheId, Balance, Rate>
+where
+	Domain: Encode + Decode,
+	PoolId: Encode + Decode,
+	TrancheId: Encode + Decode,
+	Balance: Encode + Decode,
+	Rate: Encode + Decode,
+{
+	Invalid,
+	AddPool {
+		pool_id: PoolId,
+	},
+	AddTranche {
+		pool_id: PoolId,
+		tranche_id: TrancheId,
+		token_name: [u8; 32],
+		token_symbol: [u8; 32],
+	},
+	UpdateTokenPrice {
+		pool_id: PoolId,
+		tranche_id: TrancheId,
+		price: Rate,
+	},
+	UpdateMember {
+		pool_id: PoolId,
+		tranche_id: TrancheId,
+		address: [u8; 32],
+		valid_until: Moment,
+	},
+	Transfer {
+		pool_id: PoolId,
+		tranche_id: TrancheId,
+		domain: Domain,
+		destination: [u8; 32],
+		amount: Balance,
+	},
+}
+
+impl<
+		Domain: Encode + Decode,
+		PoolId: Encode + Decode,
+		TrancheId: Encode + Decode,
+		Balance: Encode + Decode,
+		Rate: Encode + Decode,
+	> Message<Domain, PoolId, TrancheId, Balance, Rate>
+{
+	fn call_type(&self) -> u8 {
+		match self {
+			Self::Invalid => 0,
+			Self::AddPool { .. } => 1,
+			Self::AddTranche { .. } => 2,
+			Self::UpdateTokenPrice { .. } => 3,
+			Self::UpdateMember { .. } => 4,
+			Self::Transfer { .. } => 5,
+		}
```

Comment:
Consider a idiomatic way of doing this

Reply:
@lemunozm @gruberb maybe your Rust skills can come to the rescue here? I tried using `#repr(u8)` here but that's not well supported for enum variants with fields. Do you suggest some other approach?

Reply:
Hm, what does `call_type` do? Maybe adding a field `call_type` to the `enum` variants itself with an `enum CallType { NameA = 0, NameB = 1, ...}`

Reply:
`call_type` is just an `Message` variant id essentially, that needs to be stable as it's to be referenced cross-chain. Your suggestion doesn't help here since `call_type` is a value to be known at compilation-time, not at the runtime/value level. I think we can keep it as is, it's not much code and it's explicit. Thanks tho!

Reply:
There is a nightly api if I remember correctly. Bu probably not the right way to go ^^